### ISSUE TITLE
Null enum references in switch statements are not flagged

### ISSUE THREAD

**Issue by GoogleCodeExporter** [typetools/checker-framework#129]
```
The following code contains a switch statement that will throw a 
NullPointerException at runtime:

  public class Main {
    public static void main( String[] args ) {
      switch( get( ) ) {
        case X:
          System.out.println( "X" );
          break;
        default:
          System.out.println( "default" );
      }
    }

    public static /*@Nullable*/ A get( ) { return null; }

    public static enum A { X }
  }

However, the nullness checker does not flag this as an error. I tested this 
with revision 6e61ed5b3ec1 (the HEAD revision at time of writing).
```

Original issue reported on code.google.com by `atomknig...@gmail.com` on 23 Feb 2012 at 7:47
